pkg/config/util_supported.go

```diff
@@ -25,6 +25,17 @@ func getRuntimeDir() (string, error) {
 
 	rootlessRuntimeDirOnce.Do(func() {
 		runtimeDir := os.Getenv("XDG_RUNTIME_DIR")
+		if runtimeDir != "" {
+			st, err := os.Stat(runtimeDir)
+			if err != nil {
+				rootlessRuntimeDirError = err
+				return
+			}
+			if int(st.Sys().(*syscall.Stat_t).Uid) != os.Geteuid() {
```

Comment:
Shouldn't we use `uid` from line 39?

Reply:
Ah, that's a string. In other words: should we use the rootless UID?

Reply:
we need the uid inside the user namespace so `os.Geteuid()` to match against the uid returned by the stat.

The other `uid` is the original user id, so even in the user namespace, it will still keep the original value before joining it